register-styles.html

```diff
@@ -0,0 +1,126 @@
+<link rel="import" href="../polymer/lib/elements/dom-module.html">
+
+<script>
+(function() {
+
+  /**
+   * @namespace Vaadin
+   */
+  window.Vaadin = window.Vaadin || {};
+
+  let moduleIdIndex = 0;
+  // Map of <CSSResult, Polymer.DomModule> pairs.
+  const styleMap = {};
+
+  /**
+   * Registers CSS styles for a component type. Make sure to register the styles before
+   * the first instance of a component of the type is attached to DOM.
+   *
+   * @param {String} themeFor The local/tag name of the component type to register the styles for
+   * @param {CSSResult | CSSResult[]} styles The CSS style rules to be registered for the component type
+   * matching themeFor and included in the local scope of each component instance
+   * @param {Object} [options] Additional options
+   * @return {void}
+   */
+  Vaadin.registerStyles = (themeFor, styles, options) => {
+    const themeId = (options && options.moduleId) || `custom-style-module-${moduleIdIndex++}`;
+
+    if (!Array.isArray(styles)) {
+      styles = styles ? [styles] : [];
+    }
+
+    styles.forEach(cssResult => {
+      if (!(cssResult instanceof CSSResult)) {
+        throw new Error(
+          'An item in styles is not of type CSSResult. Use `unsafeCSS` or `css`.');
+      }
+      if (!styleMap[cssResult]) {
+        const styleModuleElement = document.createElement('dom-module');
+        styleModuleElement.innerHTML = `
+          <template>
+            <style>${cssResult.toString()}</style>
+          </template>
+        `;
+
+        const styleId = `custom-style-module-${moduleIdIndex++}`;
+        styleModuleElement.register(styleId);
+        styleMap[cssResult] = styleId;
+      }
+    });
+
+    const themeModuleElement = document.createElement('dom-module');
+    if (themeFor) {
+      const elementClass = window.customElements && window.customElements.get(themeFor);
+      if (elementClass && elementClass.hasOwnProperty('__finalized')) {
+        console.warn(`The custom element definition for "${themeFor}"
+        was finalized before a style module was registered.
+        Make sure to add component specific style modules before
+        importing the corresponding custom element.`);
+      }
+      themeModuleElement.setAttribute('theme-for', themeFor);
+    }
+
+    const moduleIncludes = (options && options.include) || [];
+
+    themeModuleElement.innerHTML = `
+      <template>
+        ${styles.map(style => `<style include=${styleMap[style]}></style>`)}
+        ${moduleIncludes.map(include => `<style include=${include}></style>`)}
+      </template>
+    `;
+
+    themeModuleElement.register(themeId);
+  };
+
+  /* Template literal tags derived from LitElement */
+
+  const constructionToken = Symbol();
```

Comment:
Note, `Symbol` is ES6 feature which needs a separate polyfill. That polyfill is included to `webcomponentsjs` 2.x (Polymer 3 compatible version) but not to 1.x (Polymer 2 & HTML Imports).

We probably need to change this to get Polymer 2 build passing in IE11.

Reply:
Right, changed to a plain object

Reply:
@tomivirkki Is this targeting Vaadin 14 or 15? The issue mainly talks about Polymer 3.

If we need this for V14 with IE11 support, then I guess we need to change this part, but if this is targeting only V15+ then I think using Symbol as is is fine?

Reply:
This is for V14